server/metric/mock_metric_test.go

```diff
@@ -0,0 +1,289 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metric
+
+import (
+	"context"
+	"log"
+	"net"
+	"os"
+	"testing"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"google.golang.org/genproto/googleapis/monitoring/v3"
+	"google.golang.org/genproto/googleapis/rpc/status"
+	"google.golang.org/genproto/googleapis/api/monitoredres"
+	"google.golang.org/genproto/googleapis/api/metric"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/test/bufconn"
+	"google.golang.org/protobuf/proto"
+)
+
+const bufSize = 1024 * 1024
+
+var (
+	client monitoring.MetricServiceClient
+	conn   *grpc.ClientConn
+	ctx    context.Context
+	lis    *bufconn.Listener
+)
+
+func setup() {
+	// Setup the in-memory server.
+	lis = bufconn.Listen(bufSize)
+	grpcServer := grpc.NewServer()
+	monitoring.RegisterMetricServiceServer(grpcServer, &MockMetricServer{})
+	go func() {
+		if err := grpcServer.Serve(lis); err != nil {
+			log.Fatalf("server exited with error: %v", err)
+		}
+	}()
+
+	// Setup the connection and client.
+	ctx = context.Background()
+	var err error
+	conn, err = grpc.DialContext(ctx, "bufnet", grpc.WithContextDialer(bufDialer), grpc.WithInsecure())
+	if err != nil {
+		log.Fatalf("failed to dial bufnet: %v", err)
+	}
+	client = monitoring.NewMetricServiceClient(conn)
+}
+
+func tearDown() {
+	conn.Close()
+}
+
+func TestMain(m *testing.M) {
+	setup()
+	retCode := m.Run()
+	tearDown()
+	os.Exit(retCode)
+}
+
+func bufDialer(context.Context, string) (net.Conn, error) {
+	return lis.Dial()
+}
+
+func TestMockMetricServer_CreateTimeSeries(t *testing.T) {
+	cases := []struct {
```

Comment:
Are you intending to add more test cases in a future PR? If not then would be clearer to just define `in` and `want` and not have the array and for loop.

Reply:
We haven't really decided on how/where future tests might be, but I think you're right here. For now, we're only testing the type of responses are correct, so just the in/want should be fine. We can come back later to test more functionality.